src/components/[guild]/RoleCard/components/EditRole/hooks/useEditRoleForm.ts

```diff
@@ -0,0 +1,62 @@
+import useGuild from "components/[guild]/hooks/useGuild"
+import { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import { Logic, Visibility } from "types"
+
+export type RoleEditFormData = {
+  id: number
+  name: string
+  description: string
+  imageUrl: string
+  logic: Logic
+  visibility: Visibility
+  visibilityRoleId?: number
+  anyOfNum?: number
+  groupId?: number
+}
+
+const useEditRoleForm = (roleId: number) => {
+  const { roles } = useGuild()
+  const {
+    id,
+    name,
+    description,
+    imageUrl,
+    logic,
+    anyOfNum,
+    visibility,
+    visibilityRoleId,
+    groupId,
+  } = roles?.find((role) => role.id === roleId) ?? {}
+
+  const defaultValues: RoleEditFormData = {
+    id,
+    name,
+    description,
+    imageUrl,
+    logic,
+    anyOfNum: anyOfNum ?? 1,
+    visibility,
+    visibilityRoleId,
+    groupId,
+  }
+
+  const methods = useForm<RoleEditFormData>({
+    mode: "all",
+    defaultValues,
+  })
+
+  useEffect(() => {
```

Comment:
Any idea why we have this? I know it was just copy-pasted from EditRole, maybe we could remove it now? cc @BrickheadJohnny

Reply:
This change got introduced with the visibility functionality: https://github.com/guildxyz/guild.xyz/pull/707/files#diff-70157972a09b33ce89047823bf3780ac4faea94906e5bfed1237155211b0f02dR79-R90

After thinking about it for a couple of minutes, I have no idea why would we need this. :D I think if the default form values get populated properly without this useEffect, it's safe to remove it (I guess that's the point of it?)